Core/Src/main.c

```diff
@@ -87,14 +92,40 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
-
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+
+  static const uint8_t START = 1;
+  static const uint8_t END = 2;
+  static const uint16_t MAX_ADC = 1023; // 1023 is the max value that can be stored by a 10 bit value
+
+  uint16_t adc_value = 0x0000;
+  uint16_t compare_value;
+  static const uint16_t SCALE_VALUE = 3000; // Scaling the 60000 to the period of 20ms: 60000/20=3000
+
+  HAL_GPIO_WritePin(GPIOA, GPIO_PIN_5, GPIO_PIN_SET);
+
   while (1)
   {
+	  HAL_GPIO_WritePin(GPIOA, GPIO_PIN_5, GPIO_PIN_RESET);
+
+	  HAL_SPI_TransmitReceive(&hspi1, spi_tx_buffer, spi_rx_buffer, sizeof(spi_tx_buffer), 1000);
+
+	  HAL_GPIO_WritePin(GPIOA, GPIO_PIN_5, GPIO_PIN_SET);
+
+	  adc_value = ((uint16_t) spi_rx_buffer[1] << 8) | (uint16_t) spi_rx_buffer[2];
```

Comment:
This uint16 holds the 10 data bits, however it also contains 6 more bits that are irrelevant to this program. The 10 bits we care about will need to be extracted. This can be done a couple ways, but the easiest way might be by bit masking.